tools/vddk-test/vddk-test-plugin.c

```diff
@@ -20,24 +20,27 @@
 
 #define THREAD_MODEL NBDKIT_THREAD_MODEL_SERIALIZE_ALL_REQUESTS
 
-#define EXPECTED_ARG_COUNT 7
 int arg_count = 0;
+int expected_arg_count = 7;
 
 void fakevddk_close(void *handle) {
     close(*((int *) handle));
 }
 
 int fakevddk_config(const char *key, const char *value) {
     arg_count++;
+    if (strcmp(key, "snapshot") == 0) {
+        expected_arg_count = 8;
```

Comment:
So IIUC this commit fixes the e2e test lanes, right?
What's weird is that the failures manifested themselves as rate limit errors:
```
Error: failed get datavolume: dv-vddk-warm-import-test, client rate limiter Wait returned an error: context deadline exceeded
```
Do you happen to have any insight around that?

Reply:
I don't know exactly what rate limiter is showing up here, but it makes sense that the test exceeds some deadline somewhere. I think these tests don't look at the importer pod at all, they just create a data volume and wait for it to finish. So if the importer is consistently crashing, the data volume never finishes, and the test times out.

Reply:
Right, so, in this case, the importer was crashing over this change missing?

> I don't know exactly what rate limiter is showing up here

That would be the client that tries to get said DataVolume

Reply:
Yes, this is what fixes the E2E tests. The importer was using this mock VDDK plugin, and one of the only things it checks is the number of arguments. So it was failing repeatedly without showing anything in the test logs.

Reply:
Thanks for the explanation, might bug you a little about the mock plugin offline